src/udiskslinuxmanager.c

```diff
@@ -867,6 +867,137 @@ handle_enable_modules (UDisksManager *object,
   return TRUE; /* returning TRUE means that we handled the method invocation */
 }
 
+static gboolean
+handle_can_format (UDisksManager         *object,
+                   GDBusMethodInvocation *invocation,
+                   const gchar           *type)
+{
+  gchar *required_utility = NULL;
+  gchar *binary_path = NULL;
+
+  if (g_strcmp0 (type, "swap") == 0)
+    required_utility = g_strdup ("mkswap");
+  if (g_strcmp0 (type, "empty") == 0)
+    required_utility = g_strdup ("wipefs");
+  else
+    {
+      const gchar **supported_fs = get_supported_filesystems ();
+      for (gsize i = 0; supported_fs[i] != NULL; i++)
+        {
+          if (g_strcmp0 (type, supported_fs[i]) == 0)
+            {
+            required_utility = g_strconcat ("mkfs.", type, NULL);
+            break;
+            }
+        }
+    }
+
+  if (required_utility == NULL)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                     UDISKS_ERROR,
+                     UDISKS_ERROR_NOT_SUPPORTED,
+                     "Creation of filesystem type %s is not supported",
+                     type);
+      return TRUE;
+    }
+
+  binary_path = g_find_program_in_path (required_utility);
+  udisks_manager_complete_can_format (object,
+                                      invocation,
+                                      g_variant_new ("(bs)", binary_path != NULL,
+                                                     binary_path != NULL ? "" : required_utility));
+  g_free (binary_path);
+  g_free (required_utility);
+
+  return TRUE;
+}
+
+static gboolean
+handle_can_resize (UDisksManager         *object,
+                   GDBusMethodInvocation *invocation,
+                   const gchar           *type)
+{
+  GError *error = NULL;
+  gchar *required_utility = NULL;
+  BDFsResizeFlags mode;
+  gboolean ret;
+
+  ret = bd_fs_can_resize (type, &mode, &required_utility, &error);
+
+  if (error != NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
```

Comment:
I believe this should use ``take_error()``, shouldn't it? The same applies below.